### PR TITLE
Refinement checking for INSTANCEs with arguments.

### PR DESCRIPTION
**Pull request by lemmy**
Addresses Github issue #687
https://github.com/tlaplus/tlaplus/issues/687

[Feature][TLC][Changelog]

### CODE REVIEW

File: tlatools/org.lamport.tlatools/src/tlc2/tool/impl/SpecProcessor.java [hunk 1]
```diff
@@ -1447,7 +1448,55 @@ private final void processConfigProps(String name, ExprNode pred, Context c, Lis
...
+					if (ctxts.isDone()) {
+						Assert.fail(EC.TLC_LIVE_FORMULA_TAUTOLOGY);
+					}
+					Context c1;
+					while ((c1 = ctxts.nextElement()) != null) {
+						final ExprNode expr = (ExprNode) args[0];
+						this.processConfigProps(expr.toString(), expr, c1, subs);
+					}
+					return;
+				} catch (Exception ignored) {
```
Comment (lemmy) [inline tlatools/org.lamport.tlatools/src/tlc2/tool/impl/SpecProcessor.java:1491]: TODO: Swallowing the exception risks that the `\A s \in ...` is only partially unrolled, leaving the property incomplete.

File: tlatools/org.lamport.tlatools/src/tlc2/tool/impl/SpecProcessor.java [hunk 2]
```diff
@@ -1447,7 +1449,65 @@ private final void processConfigProps(String name, ExprNode pred, Context c, Lis
...
+						Context c1;
+						while ((c1 = ctxts.nextElement()) != null) {
+							final ExprNode expr = (ExprNode) args[0];
+							this.processConfigProps(expr.toString(), expr, c1, subs);
+						}
+					} catch (Exception e) {
+						// Terminate immediately if any iteration of the while loop fails. 
+						// This ensures that we avoid partially unrolling the quantification, 
+						// which would leave the property incomplete.
+			            Assert.fail(EC.TLC_CONFIG_PROPERTY_NOT_CORRECTLY_DEFINED, name);
```
Comment (Calvin-L) [inline tlatools/org.lamport.tlatools/src/tlc2/tool/impl/SpecProcessor.java:1501]: Nit: odd indentation on this line (mix of tabs and spaces).

**Review by Calvin-L — APPROVED** [review]

**Comment by lemmy** [comment]
Diagnosing and testing #1244 revealed that this PR also added support to handle the following inputs:

```tla
MyNext(myNext) == myNext
SpecNext ==
  /\ Init
  /\ MyNext([][Next]_vars)
```

whereas it previously failed with:

```shell
-> % java -jar ~/TLC/tla2tools_1-7-4.jar Github1244.tla -config Github1244b.cfg
TLC2 Version 2.19 of 08 August 2024 (rev: 5a47802)
Running breadth-first search Model-Checking with fp 59 and seed -6435437547161400017 with 1 worker on 10 cores with 8192MB heap and 64MB offheap memory [pid: 56764] (Mac OS X 26.3 aarch64, Homebrew 17.0.18 x86_64, MSBDiskFPSet, DiskStateQueue).
Parsing file /Users/markus/src/TLA/tla/tlatools/org.lamport.tlatools/test-model/Github1244.tla
Semantic processing of module Github1244
Starting... (2026-02-23 11:00:50)
Computing initial states...
Finished computing initial states: 1 distinct state generated at 2026-02-23 11:00:51.
Error: No next state actions defined to generate successor states from.
Finished in 00s at (2026-02-23 11:00:51)
```

**Pull request merged**